Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -480,3 +518,17 @@ void MainWindow::SetStateSlot(int slot)
   Settings::Instance().SetStateSlot(slot);
   m_state_slot = slot;
 }
+
+bool MainWindow::eventFilter(QObject* object, QEvent* event)
+{
+  if (event->type() == QEvent::Close)
```

Comment:
By the way, what do you think of using closeEvent instead? Then you don't have to check for the event type.